src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -294,6 +294,71 @@ public Mnemonic CreateWallet(string password, string name, string passphrase, Mn
             return mnemonic;
         }
 
+        /// <inheritdoc />
+        public string SignMessage(string password, string walletName, string externalAddress, string message)
+        {
+            Guard.NotEmpty(password, nameof(password));
+            Guard.NotEmpty(walletName, nameof(walletName));
+            Guard.NotEmpty(message, nameof(message));
+            Guard.NotEmpty(externalAddress, nameof(externalAddress));
+
+            string signature = string.Empty;
+
+            // Get wallet
+            Wallet wallet = this.GetWalletByName(walletName);
+
+            // Check the password.
+            try
```

Comment:
You don't need this check. If the password isn't right, it will fail when signing the message.
You can just remove this `try...catch`.